tracing/src/main/java/com/palantir/remoting3/tracing/Trace.java

```diff
@@ -28,7 +28,7 @@
  * Represents a trace as an ordered list of non-completed spans. Supports adding and removing of spans. This class is
  * not thread-safe and is intended to be used in a thread-local context.
  */
-final class Trace {
+public final class Trace {
```

Comment:
does this need to become public?

Reply:
Without this being `public`, one of the test utils fails to compile.

<img width="1003" alt="screen shot 2018-09-06 at 17 26 13" src="https://user-images.githubusercontent.com/3473798/45171323-fcbcba00-b1f9-11e8-8ed9-2e36563bf6be.png">

I find an alternative that involved some objects and casting too!

Reply:
it is also kind of odd that we have a public static method that returns `Trace` and `Trace` is private.
https://github.com/palantir/http-remoting/pull/799/files#diff-7ee517b85058e51de91b9bf446ff7585L246

Reply:
@qinfchen I don't have the original context, but one possible justification could be that this Trace class is useful for testing but shouldn't be used by consumers??